src/fides/api/service/messaging/messaging_crud_service.py

```diff
@@ -80,32 +88,283 @@ def get_messaging_config_by_key(db: Session, key: FidesKey) -> MessagingConfigRe
     )
 
 
-def get_all_messaging_templates(db: Session) -> List[MessagingTemplate]:
-    # Retrieve all templates from the database
+def get_all_basic_messaging_templates(db: Session) -> List[MessagingTemplate]:
+    """
+    Retrieve all templates from the database, filling in default values if any default template type
+    is not found in the database.
+    """
+    #
     templates_from_db = {
-        template.key: template.content for template in MessagingTemplate.all(db)
+        template.type: template.content for template in MessagingTemplate.all(db)
     }
```

Comment:
The results of this can be confusing if there are multiple templates saved of the same type - for each template I think it will show the last template of that type -

Reply:
Good thinking. Perhaps we should use the first template instead to align with https://github.com/ethyca/fides/pull/4901#discussion_r1627904594. 

More context:

This is a "basic" template method, and the template-specific messaging feat is only avail to plus users.

This edge case would only be hit if a user enables/configures template-specific messaging, then disables it again. They would have artifact templates configured. This is problematic especially if they downgrade to OSS and cannot change the enabling/disabling of each template.

Because of this, I think we should be consistent with how we render templates in the UI and how we allow users to [create/update templates here](https://github.com/ethyca/fides/pull/4901#discussion_r1627904594).

I don't think we should use enabled/disabled since enabling/disabling by template is only avail for Plus with the ENV var enabled.

Reply:
> I don't think we should use enabled/disabled since enabling/disabling by template is only avail for Plus with the ENV var enabled.

Yep thanks for clarifying, I didn't know about this requirement!  Again any code comments around the basic templates to help future folks parse non-standard behavior would be useful.

Reply:
I've actually extracted a core "decider" method to handle this in a more consistent way across create/ get/update use cases for our Basic templates 👍